test.disabled/ejabberd_tests/tests/mod_global_distrib_SUITE.erl

```diff
@@ -854,6 +887,58 @@ redis_query(Node, Query) ->
     RedisWorker = rpc(Node, wpool_pool, best_worker, [mod_global_distrib_mapping_redis]),
     rpc(Node, eredis, q, [RedisWorker, Query]).
 
+%% A fake address we don't try to connect to.
+%% Used in test_advertised_endpoints_override_endpoints testcase.
+advertised_endpoints() ->
+    [
+     {fake_domain(), 7777}
+    ].
+
+fake_domain() ->
+    "somefakedomain.com".
+
+iptuples_to_string([]) ->
+    [];
+iptuples_to_string([{Addr, Port} | Endps]) when is_tuple(Addr) ->
+    [{inet_parse:ntoa(Addr), Port} | iptuples_to_string(Endps)];
+iptuples_to_string([E | Endps]) ->
+    [E | iptuples_to_string(Endps)].
+
+maybe_add_advertised_endpoints(NodeName, Opts, Config) ->
+    Endpoints = proplists:get_value(NodeName, ?config(add_advertised_endpoints, Config), []),
+    case Endpoints of
+        [] ->
+            Opts;
+        E ->
+            Connections = case lists:keyfind(connections, 1, Opts) of
+                              false ->
+                                  [];
+                              C -> C
+                          end,
+            NewConnections = {connections, [{advertised_endpoints, E} | Connections]},
+            [NewConnections | Opts]
+    end.
+
+load_suite_module_on_nodes() ->
+    {_Module, Binary, Filename} = code:get_object_code(?MODULE),
+    execute_on_each_node(code, load_binary, [?MODULE, Filename, Binary]).
+
+mock_inet_on_each_node() ->
+    Nodes = lists:map(fun({NodeName, _, _}) -> ct:get_config(NodeName) end, get_hosts()),
+    lists:map(fun(Node) -> rpc:block_call(Node, ?MODULE, mock_inet, []) end, Nodes).
+
+execute_on_each_node(M, F, A) ->
+    lists:map(fun({NodeName, _, _}) -> rpc(NodeName, M, F, A) end, get_hosts()).
+
+mock_inet() ->
```

Comment:
Can we have gd_inet in mongoose, which we would mock? With a bit of docs there, why it's done.
That's way we would ensure, that if meck breaks, it would break just GD, not undefined amount of other features.

Reply:
Or you can pass inet module as a parameter for mod_global_distribution.
Something like `{resolv_module, inet}` (by default). That's way you can make a separate module `inet_localhost_only` and not use meck at all.

Reply:
Tests should not require root access.
So, we can't change `/etc/hosts`.

Actually, there is one exception: I've added `muc.localhost` on travis to be able to run s2s+muc tests. But the same test runs just fine on my local machine without changing hosts, so it's most likely broken resolve function on travis.

Reply:
2nd is a general way of "Dependency injection". 
You can even specify just resolv_function.
I don't see a problem to have an extra argument. Just document the argument as for internal usage, can removed anytime.

Reply:
> In the future we are, apparently, going to get rid of all mecks

That's a contested opinion